### PR TITLE
USHIFT-367: apply rebase.sh to openshift-dns manifests

### PR DESCRIPTION
**Pull request by fzdarsky**
Updates the openshift-dns manifests using the latest rebase.sh automation.

Most changes are completely neutral and result from the adoption of `yq`, e.g. reordering of fields resulting from `yq` appending added fields at the end rather than ordering them in alphabetically or different indentation. A few changes drop comments or labels that are no longer applied by the respective operator. Most notable change is the adoption of the PSA labels for the namespace.

Signed-off-by: Frank A. Zdarsky <fzdarsky@redhat.com>

Partially addresses [USHIFT-367](https://issues.redhat.com//browse/USHIFT-367)

### CODE REVIEW

File: assets/components/openshift-dns/dns/daemonset.yaml [hunk 1]
```diff
@@ -1,101 +1,93 @@
...
-          name: metrics
-        resources:
-          requests:
-            cpu: 10m
-            memory: 40Mi
-        volumeMounts:
-        - mountPath: /etc/tls/private
-          name: metrics-tls
-          readOnly: true
+        - name: dns
```
Comment (stlaz) [inline assets/components/openshift-dns/dns/daemonset.yaml:14]: The issue with whitespaces also appears in https://github.com/openshift/microshift/pull/926

Please check that the rebase script is correct.

**Review by stlaz — COMMENTED** [review]
/hold
the script appears to be generating suspicious whitespace changes

**Comment by fzdarsky** [comment]
> /hold the script appears to be generating suspicious whitespace changes

@stlaz You're right that the indentation changes. This is a result of the rebase script editing manifests using `yq` which lints YAML with a slightly different indentation of lists (see PR description), e.g.

```
containers:
- name: dns
  ...
- name: kube-rbac-proxy
  ...
```

becomes

```
containers:
  - name: dns
    ...
  - name: kube-rbac-proxy
    ...
```

**Comment by stlaz** [comment]
/hold cancel

I see, this is explained here: https://mikefarah.gitbook.io/yq/usage/output-format

I guess this was caused either by yq change or yq integration into the script. All good from me, then.

**Comment by dhellmann** [comment]
> Updates the openshift-dns manifests using the latest rebase.sh automation.
> 
> Most changes are completely neutral and result from the adoption of `yq`, e.g. reordering of fields resulting from `yq` appending added fields at the end rather than ordering them in alphabetically or different indentation. A few changes drop comments or labels that are no longer applied by the respective operator. Most notable change is the adoption of the PSA labels for the namespace.

Are the script changes to make those edits in another PR? Or is the script already updating these manifests but the versions in this PR come from a different release image?

**Pull request merged**